### DISCUSSION TITLE
Multiple server connections to one client

### DISCUSSION THREAD

**Discussion by VictorLemosR** [rathole-org/rathole#383] [Q&A]
I'm a newbie on rust and proxy stuff, so any help is appreciated!

I have a setup where multiple servers (around 10) are connecting to a single client. Each server and client has its own configuration file. The setup looks like this:

For each server, I have separate configuration files that differ only by the service name (e.g., services.a, services.b):
```
[server]
bind_addr = "0.0.0.0:80"

[server.services.a]
token = "xxxxxxx"
bind_addr = "127.0.0.1:5000"
```
Client configuration, the files differs by 'remote_addr' and 'client.services.a':
```
[client]
remote_addr = "server1_ip:80"

[client.services.a]
token = "xxxxxxx"
local_addr = "127.0.0.1:6000"
```
I have two main questions:

1) Is there a more efficient or elegant way to manage this setup? Right now, I have to maintain 10 separate files for server and client sides, which feels redundant.

2) What would be a better approach to handling connection checks? Currently, I am polling each server until a connection is established. However, the connections are not persistent—they might be lost after some time (e.g., a day), and the server will attempt to reconnect later, which could happen as infrequently as once a day or even less. Polling every 10 seconds is problematic for some operations, but decreasing the interval to 1 second seems wasteful. Is there a more efficient way to manage this, or should I stick with frequent polling? How detrimental is this approach in practice?

**Comment by imneonizer** [comment]
checkout https://github.com/algofly-oss/porthub 
It support enrolling one client to muiltiple servers and you don't have to maintain any config files.
